### PR TITLE
feat: support scaling cluster nodes up and down

### PR DESCRIPTION
**Pull request by Unix4ever**
Also fix some nasty bugs here and there.

Signed-off-by: Artem Chernyshev <artem.chernyshev@talos-systems.com>

### CODE REVIEW

File: pkg/capi/check.go [hunk 1]
```diff
@@ -114,29 +92,15 @@ func (clusterAPI *Manager) CheckClusterReady(ctx context.Context, metalClient cl
...
-
-	machineDeployments.SetGroupVersionKind(
-		schema.GroupVersionKind{
-			Version: clusterAPI.version,
-			Group:   "cluster.x-k8s.io",
-			Kind:    "MachineDeployment",
-		},
-	)
-
-	labelSelector, err := labels.Parse("cluster.x-k8s.io/cluster-name=capi-test")
```
Comment (Unix4ever) [inline pkg/capi/check.go:131]: that was fun to discover :facepalm:

**Review by smira — APPROVED** [review]
:+1:

**Pull request merged**